tla/ccfraft.tla

```diff
@@ -481,6 +487,19 @@ AppendEntriesBatchsize(i, j) ==
      \* This can be redefined to send bigger batches of entries.
     {nextIndex[i][j]}
 
+
+PlausibleSucessorNode(i) ==
+    \* Find a plausible successor node for i
+    \* Note that while CHOOSE is deterministic, it may not pick the same
+    \* node as the implementation, which iterates through an unordered_map
+    \* when performing its selection with the same criteria.
+    LET
+        activeServers == Servers \intersect removedFromConfiguration
+        highestMatchServers == {n \in activeServers : \A m \in activeServers : matchIndex[i][n] >= matchIndex[i][m]}
+        highestConfigServers == {n \in highestMatchServers : \A m \in highestMatchServers: MaxConfigurationIndex(n) >= MaxConfigurationIndex(m)}
```

Comment:
Sorry, I'm a bit confused by this. `MaxConfigurationIndex(m)` is local state on a remote node, how does node `i` have access to this?

Reply:
The confusion is on my side, configuration is indeed a global, indexed by node. How nice it would be to have a type definition for it, and the ability to scope state by node... I'll fix this.

Reply:
I believe `HighestConfigurationWithNode()` now does the right thing, and picks, on that server (i), out of `highestMatchServers`, those in the latest configuration.

Reply:
Minor: Like most of `ccfraft.tla`, most operator names are built from the term "server" instead of "node".

Reply:
@lemmy yes, but `server` refers to the top-level index into the globals (`configurations` in this case), and this call takes two arguments, `server` (that `server`'s local view of the configurations) and `node`, which is the node we want to know about.

I think it would be a lot nicer if we could scope the state properly, so that it isn't so trivially easy to access things that aren't available to the node, but that's beyond the scope of this change (no pun intended).